precommit-pylint.py

```diff
@@ -60,53 +38,34 @@ def check_file(limit, filename, output=False):
     """
 
     # Check if file to skip
-    if os.path.basename(filename) == '__init__.py':
-        if os.stat(filename).st_size == 0:
-            print(
-                'Skipping pylint on {} (empty __init__.py..'
-                '\tSKIPPED'.format(filename))
-            return True
+    if os.path.basename(filename) == '__init__.py' and os.stat(filename).st_size == 0:
+        print(f'Skipping pylint on {filename} (empty __init__.py..\tSKIPPED')
+        return True
 
     # Start pylint
-    print('Running pylint on {}..'.format(filename), end="\t")
+    buffer = io.StringIO()
+    with contextlib.redirect_stdout(buffer):
+        with contextlib.redirect_stderr(buffer):
+            linter = pylint_run([filename], do_exit=False).linter
 
-    try:
-        command = ['pylint', filename]
-        proc = subprocess.Popen(
-            command,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE)
-
-        out, _ = proc.communicate()
-    except OSError:
-        print("\nAn error occurred. Is pylint installed?")
-        return False
+    out = buffer.getvalue()
 
     # Verify the score
-    score = _parse_score(out)
+    score = linter.stats.get('global_note', 0.0)
     ignored = _check_ignore(out)
-    if ignored or score >= float(limit):
-        status = 'PASSED'
-    else:
-        status = 'FAILED'
+    file_passed = ignored or score >= float(limit)
 
     # Add some output
-    print('{:.2}/10.00\t{}{}'.format(
-        decimal.Decimal(score),
-        status,
-        ignored and '\tIGNORED' or ''))
+    print('Running pylint on {}.. {:.2f}/10.00\t{}{}'.format(
+        filename, score,
+        'PASSED' if file_passed else 'FAILED',
+        '\tIGNORED' if ignored else ''
+    ))
 
     if output and score < 10:
```

Comment:
Looks like we do not have "output" defined...

Reply:
Output is func input param

Reply:
I run it before committing)